lib/spack/spack/test/cmd/bootstrap.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os.path
+
+import pytest
+
+import spack.config
+import spack.environment
+import spack.main
+
+_bootstrap = spack.main.SpackCommand('bootstrap')
+
+
+@pytest.mark.parametrize('scope', [
+    None, 'site', 'system', 'user'
+])
+def test_enable_and_disable(mutable_config, scope):
+    scope_args = []
+    if scope:
+        scope_args = ['--scope={0}'.format(scope)]
+
+    _bootstrap('enable', *scope_args)
+    assert spack.config.get('bootstrap:enable', scope=scope) is True
+
+    _bootstrap('disable', *scope_args)
+    assert spack.config.get('bootstrap:enable', scope=scope) is False
+
+
+@pytest.mark.parametrize('scope', [
+    None, 'site', 'system', 'user'
+])
+def test_root_get_and_set(mutable_config, scope):
+    scope_args, path = [], '/scratch/spack/bootstrap'
+    if scope:
+        scope_args = ['--scope={0}'.format(scope)]
+
+    _bootstrap('root', path, *scope_args)
+    out = _bootstrap('root', *scope_args, output=str)
```

Comment:
Don't the scope args have to come before the path? If this is passing, it makes me worried about the validity of this test.

Reply:
I don't think the position matters, since the positional argument is optional:
```console
$ spack bootstrap root "\$spack/opt/bootstrap" --scope=user
/home/culpo/PycharmProjects/spack/opt/bootstrap

$ cat ~/.spack/bootstrap.yaml 
bootstrap:
  root: $spack/opt/bootstrap
```
See also [this answer](https://stackoverflow.com/a/4480202/771663) where they parse `somedir -v`.

Reply:
Good to know it's only required positional arguments, not optional ones, that have to have the options first.